File: validate_blocks.js
const fs = require('fs');
const content = fs.readFileSync('theme/sections/main-product.liquid', 'utf-8');
const match = content.match(/{%\s*schema\s*%}([\s\S]*?){%\s*endschema\s*%}/);
if (match) {
    const schema = JSON.parse(match[1]);
    schema.blocks.forEach(block => {
        const t = block.type || '';
        const n = block.name || '';
        if (t.length > 25) console.log(`Type > 25: ${t}`);
        if (n.length > 25) console.log(`Name > 25: ${n}`);
        
        if (block.settings) {
            block.settings.forEach(s => {
                if (s.unit && s.unit.length > 3) {
                    console.log(`Unit > 3: block ${t}, unit ${s.unit}`);
                }
                if (s.type === 'range') {
                    const mx = s.max || 0;
                    const mn = s.min || 0;
                    const step = s.step || 1;
                    const dflt = s.default !== undefined ? s.default : mn;
                    if ((mx - mn) / step + 1 > 101) {
                        console.log(`Range steps > 101: block ${t}, max ${mx}, min ${mn}, step ${step}`);
                    }
                    if ((dflt - mn) % step !== 0) {
                        console.log(`Range default invalid: block ${t}, default ${dflt}, min ${mn}, step ${step}`);
                    }
                    if ((mx - mn) % step !== 0) {
                        console.log(`Range max invalid: block ${t}, max ${mx}, min ${mn}, step ${step}`);
                    }
                }
            });
        }
    });
}

File: validate_blocks.py
import json
import re

with open('theme/sections/main-product.liquid', 'r', encoding='utf-8') as f:
    content = f.read()

m = re.search(r'{%\s*schema\s*%}(.*?){%\s*endschema\s*%}', content, re.DOTALL)
if m:
    schema = json.loads(m.group(1))
    for block in schema.get('blocks', []):
        t = block.get('type', '')
        n = block.get('name', '')
        if len(t) > 25:
            print(f"Type > 25: {t}")
        if len(n) > 25:
            print(f"Name > 25: {n}")
        
        for s in block.get('settings', []):
            if 'unit' in s and len(s['unit']) > 3:
                print(f"Unit > 3: block {t}, unit {s['unit']}")
            if s.get('type') == 'range':
                mx = s.get('max', 0)
                mn = s.get('min', 0)
                step = s.get('step', 1)
                dflt = s.get('default', mn)
                if (mx - mn) / step + 1 > 101:
                    print(f"Range steps > 101: block {t}, max {mx}, min {mn}, step {step}")
                if (dflt - mn) % step != 0:
                    print(f"Range default invalid: block {t}, default {dflt}, min {mn}, step {step}")
                if (mx - mn) % step != 0:
                    print(f"Range max invalid: block {t}, max {mx}, min {mn}, step {step}")
